garak/cli.py

```diff
@@ -498,20 +516,39 @@ def main(arguments=None) -> None:
             command.start_run()  # start the run now that all config validation is complete
             print(f"📜 reporting to {_config.transient.report_filename}")
 
-            if parsed_specs["detector"] == []:
+            if not _config.plugins.harnesses:
+                # Set the _config.plugins.harnesses
+                if parsed_specs["detector"] == []:
+                    _config.plugins.harnesses["Probewise"] = {}
+                else:
+                    _config.plugins.harnesses["Pxd"] = {}
```

Comment:
can you just run me through the reasoning here? would this clobber harness config loaded from global / site / cli-specific config YAML?

Reply:
I suspect this could be avoided. We should not clobber `config` without an explicit override from a command line flag.

If no specific harness was provided via `config` and no detectors were provided per `parsed_spec` that is the determining factor on which default harness to load when `_config.plugins.harnesses` does not contain any configuration data for a specific harness. This does expose that there may be a missing top level parameter to select a specific harness if default config we to provide for various harness types as currently finding config for `DetectorOnly` to be a selection criteria seem a bit brittle.

If there is a desire to consolidate harness selection I maybe something like:
``` Python
    harness_command = command.pxd_run
    if not _config.plugins.harnesses:
        if parsed_specs["detector"] == []:
            harness_command = command.probewise_run
    elif "DetectorOnly" in _config.plugins.harnesses:
        harness_command = command.detector_only_run
    match harness_command:
        case command.detector_only_run:
            harness_command()
        case command.probewise_run:
            harness_command(
                generator, parsed_specs["probe"], evaluator, parsed_specs["buff"]
            )
        case command.pxd_run:
            harness_command(
                    generator,
                    parsed_specs["probe"],
                    parsed_specs["detector"],
                    evaluator,
                    parsed_specs["buff"],
                    parsed_specs["buff"],
            )
        case _: # base case for invalid callable, currently not a reachable case
            logging.warn("no valid harness selected")

    command.end_run()
```

There is probably more abstraction possible here but this offers an idea.